source/common/http/http1/codec_impl.cc

```diff
@@ -421,14 +426,26 @@ void ConnectionImpl::onHeaderValue(const char* data, size_t length) {
     // Ignore trailers.
     return;
   }
+
+  const absl::string_view header_value = absl::string_view(data, length);
+
   // http-parser should filter for this
   // (https://tools.ietf.org/html/rfc7230#section-3.2.6), but it doesn't today. HeaderStrings
   // have an invariant that they must not contain embedded zero characters
   // (NUL, ASCII 0x0).
-  if (absl::string_view(data, length).find('\0') != absl::string_view::npos) {
+  if (header_value.find('\0') != absl::string_view::npos) {
     throw CodecProtocolException("http/1.1 protocol error: header value contains NUL");
   }
 
+  if (validate_headers_) {
```

Comment:
I think this can be an if-else with the above clause, since I think headerIsValid should catch the '\0'
Can you check?  It'd mean that when we remove the old code path we can remove the extra header scan without worrying about a change in functionality

Reply:
Yep, the `HeaderUtility::headerIsValid` function will catch the `\0` char, along with any other ASCII char with decimal value [0-31] (with the exception of decimal value 9, which is allowed per the RFC).